chia/cmds/cmd_classes.py

```diff
@@ -307,3 +311,178 @@ async def wallet_rpc(self, **kwargs: Any) -> AsyncIterator[WalletClientInfo]:
+            return TransactionBundle.from_bytes(file.read())
+
+
+@command_helper
+class TransactionsOut:
+    transaction_file_out: Optional[str] = option(
+        "--transaction-file-out",
+        type=str,
+        default=None,
+        help="A file to write relevant transactions to",
+        required=False,
+    )
+
+    def handle_transaction_output(self, output: list[TransactionRecord]) -> None:
+        if self.transaction_file_out is None:
+            return
+        else:
+            with open(Path(self.transaction_file_out), "wb") as file:
+                file.write(bytes(TransactionBundle(output)))
+
+
+@command_helper
+class NeedsCoinSelectionConfig:
+    min_coin_amount: CliAmount = option(
+        "-ma",
+        "--min-coin-amount",
+        "--min-amount",
+        help="Ignore coins worth less then this much XCH or CAT units",
+        type=AmountParamType(),
+        required=False,
+        default=cli_amount_none,
+    )
+    max_coin_amount: CliAmount = option(
+        "-l",
+        "--max-coin-amount",
+        "--max-amount",
+        help="Ignore coins worth more then this much XCH or CAT units",
+        type=AmountParamType(),
+        required=False,
+        default=cli_amount_none,
+    )
+    coins_to_exclude: Sequence[bytes32] = option(
+        "--exclude-coin",
+        multiple=True,
+        type=Bytes32ParamType(),
+        help="Exclude this coin from being spent.",
+    )
+    amounts_to_exclude: Sequence[CliAmount] = option(
+        "--exclude-amount",
+        multiple=True,
+        type=AmountParamType(),
+        help="Exclude any coins with this XCH or CAT amount from being included.",
+    )
+
+    def load_coin_selection_config(self, mojo_per_unit: int) -> CoinSelectionConfig:
+        return CMDCoinSelectionConfigLoader(
+            min_coin_amount=self.min_coin_amount,
+            max_coin_amount=self.max_coin_amount,
+            excluded_coin_amounts=list(_ for _ in self.amounts_to_exclude),
+            excluded_coin_ids=list(_ for _ in self.coins_to_exclude),
+        ).to_coin_selection_config(mojo_per_unit)
+
+
+@command_helper
+class NeedsTXConfig(NeedsCoinSelectionConfig):
+    reuse: Optional[bool] = option(
+        "--reuse/--new-address",
+        "--reuse-puzhash/--generate-new-puzhash",
+        help="Reuse existing address for the change.",
+        is_flag=True,
+        default=None,
+    )
+
+    def load_tx_config(self, mojo_per_unit: int, config: dict[str, Any], fingerprint: int) -> TXConfig:
+        return CMDTXConfigLoader(
+            min_coin_amount=self.min_coin_amount,
+            max_coin_amount=self.max_coin_amount,
+            excluded_coin_amounts=list(_ for _ in self.amounts_to_exclude),
+            excluded_coin_ids=list(_ for _ in self.coins_to_exclude),
+            reuse_puzhash=self.reuse,
+        ).to_tx_config(mojo_per_unit, config, fingerprint)
+
+
+_DECORATOR_APPLIED = "_DECORATOR_APPLIED"
```

Comment:
for stuff like this i often add get/set/is helper functions to spread around instead of the actual `getattar()`/`setattr()`/`hasattr()` calls.  and make it really unlikely to collide with anything else.

(providing a suggestion here, but it would at least require this be moved so would have to be applied locally)

```suggestion
_TRANSACTION_ENDPOINT_RUNNER_DECORATOR_APPLIED_ATTRIBUTE_NAME = f"_{__name__.split('.')}_{transaction_endpoint_runner.__qualname__}_applied"
```

Reply:
taken